### ISSUE TITLE
Circumvent security pop-up in macOS Mojave?

### ISSUE THREAD

**Issue by frankrolf** [rkanter/Open-iTerm-or-Terminal-Here...#1]
<img width="532" alt="screen shot 2018-10-06 at 14 45 33" src="https://user-images.githubusercontent.com/2049645/46568883-8e912180-c976-11e8-8adb-d7291d488477.png">

Many AppleScript tools have this problem – any suggestions on how to circumvent this pop-up window on every single launch? I have tried these steps to no obvious effect: https://apple.stackexchange.com/questions/335848/mojave-disable-or-control-the-some-app-name-wants-access-to-control-another

Thanks!

**Comment by tdegeus** [comment]
Same here, it would be great to hear updates on this issue.

**Comment by rkanter** [comment]
Unfortunately, I don't currently have access to a computer with Mojave at the moment to look into this; I may be able to in a few weeks.

It sounds like you're supposed to be able to click "OK" or add the app to the "Full Disk Access" in the System Prefs, but that's not working consistently.  

From Googling around a bit about this OS X change, it sounds like a big problem for anything using Applescript.  And there's not a great solution.  I'm not sure how Apple was expecting people to deal with this...

But I'll definitely take a look once I get a computer with Mojave.  In the meantime, if anyone figures out a solution, feel free to post a pull request.

**Comment by tdegeus** [comment]
@rkanter Thanks for the update! I did click OK, and consequently the script works fine. However, this message keeps reappearing. This is different with some other apps, where the change is permanent and I can see this in system preferences. For example for Sublime Text:

<img width="598" alt="screenshot 2018-10-17 at 08 26 37" src="https://user-images.githubusercontent.com/7030445/47066456-77f49100-d1e6-11e8-9d94-740c3e2c9776.png">

Unfortunately I do not see a way to permanently allow it, because `Open-iTerm-or-Terminal-Here...` does not show up in system preferences.

**Comment by frankrolf** [comment]
Related, and discouraging:
https://latenightsw.com/mojave-and-applescript-applets/
https://latenightsw.com/mojave-brings-in-big-security-changes/

**Comment by rkanter** [comment]
Those articles make it sound like once you've approved it, it shouldn't need to ask you as long as the script isn't modified (and that's consistent with what I've read before).  If accurate, as long as you're not always regenerating the .app, it should be a one-time annoyance.  Is it really asking every time?

**Comment by tdegeus** [comment]
Yes, for me it is asking every time. (Like I said before, one thing to note is that the App does not show up in the security settings, indicating that the system did not remember).

**Comment by rkanter** [comment]
I thought I'd be getting Mojave on my work laptop, but they're still blocking it, so I haven't been able to try this out still.  I'm sure we'll get Mojave eventually, but if someone figures something out in the meantime, feel free to submit a pull request or post useful info here.

**Comment by frankrolf** [comment]
I have a workaround, which sadly includes getting rid of the AppleScript applet altogether:
– add iTerm or Terminal (the proper application) to the top of the Finder window (cmd-drag)
– the current folder can be drag-dropped from the top bar onto that icon

This reproduces the “iTerm here” functionality, but unfortunately is not as quick.

**Comment by rkanter** [comment]
I finally upgraded to Mojave the other day and also ran into this issue.  Though it's inconsistent... Sometimes it'll ask me, sometimes it'll just work silently, and sometimes it'll give me a weird error.

However, I may have found a simple workaround (and possibly the cause)!
From this [thread](https://forum.latenightsw.com/t/mojave-and-applescript-applets/1563/12), it sounds like the problem (or at least one of the problems) may be that the script is changing, and so Mojave sees that (and it's not signed).  I did some checking, and I'm not sure why, but it seems that the file does indeed randomly change sometimes when I run it, and that coincides with it prompting me.  Maybe AppleScript is changing some metadata or something when it runs sometimes?  Seems like a weird thing to do...
Anyway, here's some proof:
````
$ pwd
/Applications/Open iTerm Here....app/Contents/Resources/Scripts
$ ls -l && md5sum main.scpt
total 8
-rw-r--r--  1 robert  admin  4046 Jan 24 00:02 main.scpt
cf234c5b9c1be2ef03cf261eae81bad6  main.scpt
$ ls -l && md5sum main.scpt
total 8
-rw-r--r--  1 robert  admin  3946 Jan 24 00:02 main.scpt
efc4e5d1355c33e274ccfe4b9e909e9c  main.scpt
````
As you can see, the md5 and even the filesize changed.  Plus, the mod time is very recent, even though I last generated the app months ago (all of the other files in the .app bundle are from back then - the script file is the only one that's been updated).  

I tried changing the permissions of the script in the app to read only so it won't change anymore, and that seems to have fixed it.  I'll play around with it for a bit longer though.
````
$ chmod 444 main.scpt
$ ls -l
total 8
-r--r--r--  1 robert  admin  3946 Jan 24 00:02 main.scpt
````
In the meantime, if anyone else could try this, that would be helpful.  It's hard to test something that otherwise acts inconsistently :/

**Comment by tdegeus** [comment]
So far so good! I will post here should the work-around fail.

**Comment by rkanter** [comment]
Looks like everything works, except "Open Terminal Here..." doesn't work if the app was generated on Mojave and Terminal is already running.  In that case, the AppleScript sends a key combo (CMD+T) to open a new tab, and Mojave doesn't allow that, even if you add it to Accessibility (where it's simply called "applet").  I'll see if I can find an alternative or workaround for this.

**Comment by rkanter** [comment]
I figured out a fix for that issue too.
I'll try to commit an update with the fixes (and some explanation in the README) sometime in the next few days.

**Comment by frankrolf** [comment]
`chmod 444 main.scpt` indeed brings the old behavior back! Thank you very much!

**Comment by rkanter** [comment]
This is now fixed with commit [aff29805ea03a2b93a3c3f5ef3bd52c99a7dee6f](https://github.com/rkanter/Open-iTerm-or-Terminal-Here.../commit/aff29805ea03a2b93a3c3f5ef3bd52c99a7dee6f).

If you run the generate.sh script again, it should update the app with the fixes.  You'll see the prompts the first time you run it, but it should work correctly after that.  And for the Terminal version, there's an extra step to add to add it to the "Accessibility" section.  More details with screenshots are described on [MOJAVE.md](https://github.com/rkanter/Open-iTerm-or-Terminal-Here.../blob/master/MOJAVE.md).

**Comment by ElhemEnohpi** [comment]
@rkanter , fyi you don't need to make the script read-only. Just declare the variables as local, i.e., add this to the top of the iTerm script:
`local thePath, theName, createdWindow`
and to the Terminal script just:
`local thePath, theName`
